src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs

```diff
@@ -71,5 +72,32 @@ public static void AssertRoundtrips<T>(T expected, params Func<T, object>[] addi
                 }
             }
         }
+
+        public static void AssertExceptionDeserializationFails<T>() where T : Exception
+        {
+            // .NET Core throws a PlatformNotSupportedException for deserializing many exceptions.
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Core"))
+            {
+                return;
+            }
+
+            var info = new SerializationInfo(typeof(T), new FormatterConverter());
```

Comment:
Why even bothering with creating a full SerializationInfo bucket when it will instantly fail at invoking the Constructor?